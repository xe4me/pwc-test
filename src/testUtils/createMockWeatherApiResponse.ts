export const createMockWeatherApiResponse = (city_name: string) => ({
    city_name,
    data: [
      {
        wind_cdir: 'W',
        pres: 1014,
        high_temp: 20.2,
        wind_gust_spd: 5.9,
        wind_spd: 3.1,
        dewpt: 5.9,
        weather: {
          icon: 'c02d',
          code: 801,
          description: 'Few clouds',
        },
        wind_dir: 266,
        precip: 0,
        low_temp: 9.3,
        max_temp: 20.2,
        datetime: '2022-06-25',
        temp: 14.9,
        min_temp: 9.9,
      },
    ],
  });
