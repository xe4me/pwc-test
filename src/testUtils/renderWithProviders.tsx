import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material';
import theme from '../theme';

export const renderWithProviders = (element: ReactElement) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
