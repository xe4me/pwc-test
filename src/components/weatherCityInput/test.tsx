import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import { renderWithProviders } from 'testUtils/renderWithProviders';
import { axe } from 'jest-axe';
import { WeatherCityInput } from './index';

const renderComponent = (onUnitChangeSpy: any = jest.fn()) => renderWithProviders(
    <WeatherCityInput city="Sydney" onChange={onUnitChangeSpy} />
  );
describe('<WeatherCityInput>', () => {
  describe('Snapshot and accessibility', () => {
    it('should match with the snapshot', () => {
      const { baseElement } = renderComponent();
      expect(baseElement).toMatchSnapshot();
    });
    it('should pass accessibility tests', async () => {
      const html = renderComponent();
      expect(await axe(html.container)).toHaveNoViolations();
    });
  });
  describe('Visible Elements', () => {
    it('should have a label role as City input', () => {
      renderComponent();
      expect(screen.queryByLabelText('City input')).toBeInTheDocument();
    });
    it('should have a button with label role as Search', () => {
      renderComponent();
      expect(screen.queryByLabelText('Search button')).toBeInTheDocument();
    });
    it('should have a label as City', () => {
      renderComponent();
      expect(screen.queryByText('City')).toBeInTheDocument();
    });
    it('should render the given city ', () => {
      renderComponent();
      expect(screen.queryByDisplayValue('Sydney')).toBeInTheDocument();
    });
  });
  describe('City input behaviours', () => {
    it('should have a label role as City input', async () => {
      renderComponent();
      expect(screen.queryByLabelText('City input')).toBeInTheDocument();
      await fireEvent.change(screen.getByLabelText('City input'), {
        target: { value: 'hello' },
      });
      expect(screen.queryByDisplayValue('hello')).toBeInTheDocument();
    });
    it('should fire a change event after search button is clicked', async () => {
      const onChangeSpy = jest.fn();
      renderComponent(onChangeSpy);
      expect(screen.queryByLabelText('City input')).toBeInTheDocument();
      await fireEvent.change(screen.getByLabelText('City input'), {
        target: { value: 'hello' },
      });
      expect(onChangeSpy).not.toHaveBeenCalled();
      await fireEvent.click(screen.getByLabelText('Search button'));
      expect(onChangeSpy).toHaveBeenCalledWith('hello');
    });
  });
});
