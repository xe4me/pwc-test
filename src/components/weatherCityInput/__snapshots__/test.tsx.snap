// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherCityInput> Snapshot and accessibility should match with the snapshot 1`] = `
<body>
  <div>
    <form
      data-testid="weather-city-input"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-s8p2ph-MuiPaper-root"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-u4tvz2-MuiFormLabel-root"
            id="weather-city-input"
          >
            City
          </label>
          <div
            aria-labelledby="weather-city-input"
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1le28he-MuiInputBase-root"
          >
            <input
              aria-invalid="false"
              aria-label="City input"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              placeholder="Enter a city in NSW"
              type="text"
              value="Sydney"
            />
          </div>
        </div>
        <button
          aria-label="Search button"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</body>
`;
